### PR TITLE
tesselate fill paths

### PR DESCRIPTION
**Pull request by jonahwilliams**
Requires a tesselator.dll or dylib.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
@dnfield

File: packages/compiler/flutter_logo.svg [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        <linearGradient id="triangleGradient">
+            <stop offset="20%" stop-color="#000000" stop-opacity=".55" />
+            <stop offset="85%" stop-color="#616161" stop-opacity=".01" />
+        </linearGradient>
+        <linearGradient id="rectangleGradient" x1="0%" x2="0%" y1="0%" y2="100%">
+            <stop offset="20%" stop-color="#000000" stop-opacity=".15" />
+            <stop offset="85%" stop-color="#616161" stop-opacity=".01" />
+        </linearGradient>
+    </defs>
+    <path fill="#42A5F5" fill-opacity=".8" d="M37.7 128.9 9.8 101 100.4 10.4 156.2 10.4"/>
```
Comment (jonahwilliams) [inline packages/compiler/flutter_logo.svg:12]: Doesn't render correctly yet since we need to serialize the shader info, but shapes are correct!

Comment (dnfield) [reply packages/compiler/flutter_logo.svg:12]: Neat. The gradient stuff shouldn't be too hard to encode.

File: packages/compiler/lib/src/encoder.dart [hunk 2]
```diff
@@ -191,11 +194,18 @@ class PaintingCodec extends StandardMessageCodec {
   }
 
   void _writeDrawCommand(WriteBuffer buffer, DrawCommand command) {
-    buffer.putUint8(_drawCommandTag);
-    assert(_pathIds.containsKey(command.path), 'Expected to find ${command.path.hashCode}, have ${_pathIds.keys.map((p) => p.hashCode).toList()}');
-    assert(_paintIds.containsKey(command.paint));
-    buffer.putInt32(_pathIds[command.path]!);
-    buffer.putInt32(_paintIds[command.paint]!);
+    if (command is DrawPathCommand) {
```
Comment (dnfield) [inline packages/compiler/lib/src/encoder.dart:197]: nit: we may want to have an enumeration of these as a separate field so we can be more confident about switch statements having all the cases, i.e. `enum DrawCommandType { path, vertices }`

**Review by dnfield — APPROVED** [review]

**Comment by jonahwilliams** [comment]
image

**Pull request merged**